pkg/plugins/backendplugin/manager/plugin_settings.go

```diff
@@ -10,8 +11,13 @@ import (
 type pluginSettings map[string]string
 
 func (ps pluginSettings) ToEnv(prefix string, hostEnv []string) []string {
-	env := []string{}
+	var env []string
 	for k, v := range ps {
+		key := fmt.Sprintf("%s_%s", prefix, strings.ToUpper(k))
+		if value := os.Getenv(key); value != "" {
+			v = value
+		}
+
 		env = append(env, fmt.Sprintf("%s_%s=%s", prefix, strings.ToUpper(k), v))
```

Comment:
You can also change this to the following for more readable code:

```suggestion
		env = append(env, fmt.Sprintf("%s=%s", key, v))
```